site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -25,17 +29,22 @@
 @Service
 public class SiteServiceImpl implements SiteService {
 
+    private final Map<String, SiteConfiguration> sitesByType = new HashMap<>();
+    private final Map<String, String> parentTypeByChildType = new HashMap<>();
     private final EntityStore<Site, String> siteStore;
     private final ModelEntityValidation<Site> entityValidation;
     private final Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
 
     /**
+     * @param configurationProvider provide Site Configuration
      * @param siteStore        Site store interface
      * @param entityValidation Site validation interface
      */
     @Autowired
-    public SiteServiceImpl(final EntityStore<Site, String> siteStore,
+    public SiteServiceImpl(SiteConfigurationProvider configurationProvider,
+                           final EntityStore<Site, String> siteStore,
                            ModelEntityValidation<Site> entityValidation) {
+        var configuration = configurationProvider.getConfiguration();
```

Comment:
Why no null check on configurationProvider and configurationProvider.getConfiguration()?

Reply:
Will add the null chceck